engine.go

```diff
@@ -445,6 +446,10 @@ func (e *Engine) beginTransaction(ctx *sql.Context, parsed sql.Node) (string, er
 				if err != nil {
 					return "", err
 				}
+				// Replace savepoint if existing
+				tdb.ReleaseSavepoint(ctx, tx, "__go_mysql_server_starting_savepoint__")
```

Comment:
This isn't good enough: ACID applies at a statement level in addition to the transaction level. Within a single transaction, any trigger that fails to apply cleanly must roll the session back to the state it began in. Write tests for this.